internal/ethapi/api_test.go

```diff
@@ -2062,3 +2067,102 @@ func testRPCResponseWithFile(t *testing.T, testid int, result interface{}, rpc s
 	}
 	require.JSONEqf(t, string(want), string(data), "test %d: json not match, want: %s, have: %s", testid, string(want), string(data))
 }
+
+func TestBlockChainAPI_isAllowedToBeQueried(t *testing.T) {
+	t.Parallel()
+
+	const recentBlocksWindow uint64 = 1024
+
+	makeBlockWithNumber := func(number uint64) *types.Block {
+		header := &types.Header{
+			Number: big.NewInt(int64(number)),
+		}
+		return types.NewBlock(header, nil, nil, nil, nil)
+	}
+
+	testCases := map[string]struct {
+		blockNumOrHash rpc.BlockNumberOrHash
+		makeBackend    func(ctrl *gomock.Controller) *MockBackend
+		wantErrMessage string
+	}{
+		"historical_proofs_accepted": {
+			blockNumOrHash: rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(0)),
+			makeBackend: func(ctrl *gomock.Controller) *MockBackend {
+				backend := NewMockBackend(ctrl)
+				backend.EXPECT().HistoricalConfig().Return(true, recentBlocksWindow)
```

Comment:
I'm not too opinionated on this and defer to others.
Subjectively I would slightly prefer to use `testBackend` since it's already there.
In any case the new mocks need to be listed in `scripts/mocks.mockgen.txt`